message/message-util/util/src/java/org/sakaiproject/message/util/BaseMessage.java

```diff
@@ -3006,18 +3046,23 @@ protected Message findMessage(String messageId)
 		 * 
 		 * @return a List of all messages in the channel.
 		 */
-		protected List findMessages()
-		{
-			// if we have done this already in this thread, use that
-			List msgs = (List) m_threadLocalManager.get(getReference() + ".msgs");
-			if (msgs == null)
-			{
-				msgs = m_storage.getMessages(this);
-
-				// "cache" the mesasge in the current service in case they are needed again in this thread...
-				m_threadLocalManager.set(getReference() + ".msgs", msgs);
+		protected List findMessages() {
+			List msgs;
+			final List<Message> cachedMessages = messagesCache.get(getReference());
+			if (cachedMessages != null) {
+				msgs = cachedMessages;
+			} else {
+				// if we have done this already in this thread, use that
+				msgs = (List) m_threadLocalManager.get(getReference() + ".msgs");
```

Comment:
can we remove the use of threadlocalmanager here? this shouldn't be needed if were caching?
this is not an ideal pattern for passing data around.
just call `m_storage.getMessages(this)`

Reply:
Done!